adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_GETMethodBodyTest.java

```diff
@@ -116,30 +123,105 @@ public void shouldPassHttpHeadersToRestProcessor() throws Exception {
+                                                .withResponseType("application/vnd.ctx.query.somemediatype3+json"))
+                                        .withResponseTypes(
+                                                "application/vnd.ctx.query.somemediatype1+json",
+                                                "application/vnd.ctx.query.somemediatype2+json",
+                                                "application/vnd.ctx.query.somemediatype3+json"))
+                ).build(),
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, ImmutableMap.of("actionMapping", "true")));
+
+        Class<?> resourceClass = compiler.compiledClassOf(BASE_PACKAGE, "resource", "DefaultUserResource");
+        Object resourceObject = instanceOf(resourceClass);
+
+        Class<?> actionMapperClass = compiler.compiledClassOf(BASE_PACKAGE, "mapper", "DefaultUserResourceActionMapper");
+        Object actionMapperObject = actionMapperClass.newInstance();
+        setField(resourceObject, "actionMapper", actionMapperObject);
+
+        setField(resourceObject, "headers", headersWith("Accept", "application/vnd.ctx.query.somemediatype1+json"));
+
+        Method method = firstMethodOf(resourceClass);
+        method.invoke(resourceObject);
+
+
+        verify(restProcessor).processSynchronously(any(Function.class), eq("contextA.someAction"),
+                any(HttpHeaders.class), any(Map.class));
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldPassActionToRestProcessor2() throws Exception {
+        generator.run(
+                restRamlWithDefaults().with(
+                        resource("/case")
+                                .with(httpAction(GET)
+                                        .with(mapping()
+                                                .withName("contextB.action1")
+                                                .withResponseType("application/vnd.ctx.query.mediatype1+json"))
+                                        .withResponseTypes("application/vnd.ctx.query.mediatype1+json"))
+                ).build(),
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, ImmutableMap.of("actionMapping", "true")));
+
+        Class<?> resourceClass = compiler.compiledClassOf(BASE_PACKAGE, "resource", "DefaultCaseResource");
+        Object resourceObject = instanceOf(resourceClass);
+
+
+        Class<?> actionMapperClass = compiler.compiledClassOf(BASE_PACKAGE, "mapper", "DefaultCaseResourceActionMapper");
+        Object actionMapperObject = actionMapperClass.newInstance();
+        setField(resourceObject, "actionMapper", actionMapperObject);
+
+        setField(resourceObject, "headers", headersWith("Accept", "application/vnd.ctx.query.mediatype1+json"));
+
+        Method method = firstMethodOf(resourceClass);
+        method.invoke(resourceObject);
+
+
+        verify(restProcessor).processSynchronously(any(Function.class), eq("contextB.action1"),
```

Comment:
two tests, "some action" and "action1" - can we keep these names consistent action1, action2 etc for clarity?